packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -590,9 +712,19 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
 
   // The Rect representing the child's parent.
   Rect get _viewport {
+    /*
     assert(_parentKey.currentContext != null);
     final RenderBox parentRenderBox = _parentKey.currentContext!.findRenderObject()! as RenderBox;
     return Offset.zero & parentRenderBox.size;
+    */
+    assert(_parentKey.currentContext != null);
+    if (_parentKey.currentContext != null) {
+      final RenderBox parentRenderBox = _parentKey.currentContext!.findRenderObject()! as RenderBox;
+      if (parentRenderBox.hasSize) {
+        return Offset.zero & parentRenderBox.size;
+      }
+    }
+    return Offset.zero & _maxConstraints;
```

Comment:
I was previously using _parentKey to get the size of the viewport, but now I need it before the first render.  It seems like using LayoutBuilder's max constraints should work just as well...  Is there any case where that won't work?

By "viewport" I mean the size of the InteractiveViewer widget.

Reply:
Sounds reasonable. Instead of storing it in a field, can you just turn this into a method and pass it in from the build method?

Reply:
Actually I've realized that the constraints-based calculation will be wrong when InteractiveViewer is sized by its child, i.e. when the child is smaller than the constraints.  This is the typical problem of knowing the size one frame too late.

I think I've worked around it by requiring InteractiveViewer.builder to be used with `constrained: false`.  That will make this constraints-based viewport calculation correct, and I don't think there's a use case for needing `constrained` to be true with builder.

If you've got another idea let me know, though.